addons/knowledge/models/knowledge_article.py

```diff
@@ -406,6 +422,10 @@ def action_set_unlock(self):
         for article in self:
             article.is_locked = False
 
+    def action_archive(self):
+        self.child_ids.write({'active': False})
+        return super().action_archive()
```

Comment:
we have to call action_archive for all the children so that is recursive for all the descendants.
and finally archive self.
But as you will need to make a function to get all the descendants of an article, maybe use that function here.
(self | get_descendants()).write({'acitve': False})